pkg/analysis/passes/logos/logos.go

```diff
@@ -14,36 +14,41 @@ var (
 
 var Analyzer = &analysis.Analyzer{
 	Name:     "logos",
-	Requires: []*analysis.Analyzer{metadata.Analyzer},
+	Requires: []*analysis.Analyzer{nestedmetadata.Analyzer},
 	Run:      run,
 	Rules:    []*analysis.Rule{logos},
 }
 
 func run(pass *analysis.Pass) (interface{}, error) {
-	metadataBody, ok := pass.ResultOf[metadata.Analyzer].([]byte)
+
+	metadatamap, ok := pass.ResultOf[nestedmetadata.Analyzer].(nestedmetadata.Metadatamap)
 	if !ok {
 		return nil, nil
 	}
 
-	var data metadata.Metadata
-	if err := json.Unmarshal(metadataBody, &data); err != nil {
-		return nil, err
-	}
-
 	reportCount := 0
-	if strings.TrimSpace(data.Info.Logos.Small) == "" {
-		reportCount++
-		pass.ReportResult(pass.AnalyzerName, logos, "plugin.json: invalid empty small logo path", "Logo path cannot be empty")
+	for key, data := range metadatamap {
```

Comment:
the biggest change is this loop.

now instead of validating the single plugin.json provided by the metadata validator, it loops through the nestedmetadata and performs the same checks.